### ISSUE TITLE
[a11y]: DatePicker [Range With Calendar] - icon trigger bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15255]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.39.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
Calender icon that triggers the Calendar widget is not programatically identified

The Calendar icon is not keyboard navigable and not identified by the screen reader. Hence the user would not know its presence till he does down arrow from within the input field.

### Recommended Fix
The Calendar control can be made SR and keyboard reachable so that it would be apparently clear for the user.

When the screen reader doesn’t identify the page elements, it impacts the experience of users who are blind or have vision impairment. They can miss important information or fail to perform an action on the web page. 

### WCAG 2.1 Violation
4.1.2 Name, Role, Value
